package NYT.NCellMaster.NProto;

////////////////////////////////////////////////////////////////////////////////

message TCellStatistics
{
    required int64 chunk_count = 1;
    required int64 lost_vital_chunk_count = 2;
    optional int32 online_node_count = 3;

    optional int64 data_missing_chunk_count = 4;
    optional int64 parity_missing_chunk_count = 5;

    optional int64 oldest_part_missing_chunk_count = 6;
    optional int64 quorum_missing_chunk_count = 7;
    optional int64 inconsistently_placed_chunk_count = 8;
}

message TReqSetCellStatistics
{
    required uint32 cell_tag = 1;
    required TCellStatistics statistics = 2;
}

message TReqSetMulticellStatistics
{
    repeated TReqSetCellStatistics statistics = 1;
}

////////////////////////////////////////////////////////////////////////////////
